pages/LoginView.neu.page.ts

```diff
@@ -0,0 +1,42 @@
+import { expect, Page } from '@playwright/test';
+import generator from 'generate-password-ts';
+import { StartViewPage } from './StartView.neu.page';
+
+export class LoginViewPage {
+  /* add global locators here */
+  readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  /* actions */
+  public async waitForPageLoad(): Promise<void> {
+    await this.page.getByTestId('login-page-title').waitFor({ state: 'visible' });
+    await expect(this.page.getByTestId('login-page-title')).toHaveText('Anmeldung');
+  }
+
+  async login(
+    username: string = process.env.USER as string,
+    password: string = process.env.PW as string
+  ): Promise<StartViewPage> {
+    await expect(this.page.getByTestId('login-page-title')).toHaveText('Anmeldung');
+    await expect(this.page.getByTestId('login-prompt-text')).toHaveText('Bitte geben Sie Ihre persönlichen Zugangsdaten ein.');
+    await this.page.getByTestId('username-input').fill(username);
+    await this.page.getByTestId('password-input').fill(password);
+    await this.page.getByTestId('login-button').click();
+    return new StartViewPage(this.page);
+  }
+
+  async updatePassword(): Promise<string> {
+    const newPassword: string = generator.generate({ length: 8, numbers: true }) + '1Aa!';
+    await expect(this.page.getByTestId('update-password-title')).toHaveText('Passwort festlegen');
+    await expect(this.page.locator('.password-update-prompt')).toHaveText('Bitte legen Sie ein neues, selbstgewähltes Passwort fest.');
+    await this.page.getByTestId('new-password-input').fill(newPassword);
+    await this.page.getByTestId('new-password-confirm-input').fill(newPassword);
+    await this.page.getByTestId('set-password-button').click();
```

Comment:
```suggestion
  const newPasswordInput = this.page.getByTestId('new-password-input');
  const newPasswordConfirmInput = this.page.getByTestId('new-password-confirm-input');
  const setPasswordButton = this.page.getByTestId('set-password-button');

  await newPasswordInput.waitFor({ state: 'visible' });
  await newPasswordInput.fill(newPassword);

  await newPasswordConfirmInput.waitFor({ state: 'visible' });
  await newPasswordConfirmInput.fill(newPassword);

  await setPasswordButton.waitFor({ state: 'visible' });
  await setPasswordButton.click();
```
Safer maybe to make sure those inputs exist before filling? If you think this makes sense you can do it for all other pages, otherwise ignore this comment.

Reply:
I will go and check other forms for inputs and buttons and wait for them to be visible. If the locators are used on a global scope, I will define them at top class level